File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.js
function solution(k, m, score) {
    let answer = 0
    score.sort((a,b) => b - a)
    
    let box = []
    let rlt = []
    function cal(box) {
        let min_num = Math.min(...box)
        rlt.push(min_num*box.length)
    }
    
    for ( let apple of score) {
        if ( box.length == m ) {
            cal(box)
            box = [apple]
        } else { box.push(apple) }
    }
    
    if ( box.length === m ) { cal(box) }
    
    answer = rlt.reduce((a,b) => a+b,0)
    
    
    return answer
}

File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    answer = 0
    
    box = []
    rlt = []
    
    score.sort(reverse = True)
    
    def cal(box):
        min_num = min(box)
        rlt.append(min_num*len(box))
        
    for apple in score:
        if len(box) == m:
            cal(box)
            box = [apple]
        else:
            box.append(apple)
    
    if len(box) == m:
        cal(box)
    
    answer = sum(rlt)
    return answer
